pydeequ/checks.py

```diff
@@ -116,6 +116,14 @@ def addConstraint(self, constraint):
         self.constraints.append(constraint)
         self._Check = constraint._Check
 
+    def where(self, filter: str):
+        is_filterable = self._jvm.py4j.reflection.TypeUtil.isInstanceOf(
+            "com.amazon.deequ.checks.CheckWithLastConstraintFilterable", self._Check)
+        if not is_filterable:
+            raise TypeError(f"Expected class CheckWithLastConstraintFilterable, not {self._Check.getClass()}")
+        self._Check = self._Check.where(filter)
```

Comment:
I think generally most if not all checks are already wrapped by `CheckWithLastConstraintFilterable` in Deequ (I find some `kllSketchSatisfies` that is not filterable but it seems an edge case). 

Is it easier to capture a `py4j.Py4JException` and parse exception message instead?  It should have something like `method does not exist` etc. I would like to avoid reflection in this case.

Reply:
There are multiple cases in which you can't apply `.where` such as: empty Check, already applied where method once or method that doesn't return `CheckWithLastConstraintFilterable`. Another way to avoid reflection is to `.getClass()` and check if it `endswith` `CheckWithLastConstraintFilterable `.

We can definitely capture exception and the message is going to look like that: `Method where([class java.lang.String]) does not exist`, but I tried to control flow with if statement rather than by catching exceptions. Both solutions work, I will add more tests soon.

Reply:
Sounds good - I think some intuitive error message should help,  like "where filter is not supported when x,y,z: <detailed message}"

Reply:
So do you prefer the option with `endswith` or exception catching?

Reply:
I think exception catching is better, with combined with the class name in error message.

Reply:
Updated